### ISSUE TITLE
QR code based communication improvements 

### ISSUE THREAD

**Issue by antuz123** [bithyve/hexa#3337]
F&F request 

Requester: 
1. Add a contact 
2. Associate a contact (skip possible here) 
3. A QR code of an encryption key (EK) is generated and shown
4. A location ID is generated based on the EK and the request data is encrypted and uploaded on the relay at that location while the QR is being scanned. The channel flag is active from their side and new flag is set to true 
5. The requester confirms that the QR has been scanned 'Done' 
6. The app checks for active flag from the contact's side on the channel 
7. A success toast message is shown on the F&F screen where the contact is now present 

Approver: 
4.1. Scans the QR to receive the loc id and EK
4.2. Fetches the data on the loc id and decrypts 
4.3. Checks the version number, wallet name, request type etc from the channel 
4.4. Approves the request and starts uploading their data on the channel, sets channel as active from their side, new data flag as true and triggers a FCM notification for the other contact. Also start syncing the data that needs to be synced (downloaded) and set the new flag from the requester as false 
4.5. While 4.4 is happening: Associates a contact (skip possible here) 
4.6. A success toast message is shown on the F&F screen where the contact is present

Exception paths:
5. If done is not pressed, simply go to thee F&F screen and skip 6 and 7. This will get checked next time when the user refreshes the F&F screen or logins and finds the data flag on the channel 
6. If the data is not found on the channel yet, do thee same as above 
4.4 if declined, the app puts an inactive flag on the channel 

- QR Type to be specified in QR data
- QR code image should have Hexa Logo and QR Type

**Comment by Parsh** [comment]
Following have not been done out of the above:

Requester:
5. The requester confirms that the QR has been scanned 'Done'
6. The app checks for the active flag from the contact's side on the channel

The above two happen on the pull-down of the F&F screen. Would be done once the new UI is in place.

Approver:
4.4. FCM notification isn't being sent yet. Would be in place once the message center is ready.

QR code image should have Hexa Logo and QR Type (would need some assistance with the UI work)

**Comment by antuz123** [comment]
I suggest you make a new issue of the same and link it to this one. Put the next issue in 180

**Comment by cakesoft-swati** [comment]
Clsing this issue as discussed with dev.Trusted contact flow is working.
